compiler/passes/normalize.cpp

```diff
@@ -1293,14 +1301,163 @@ static bool isVoidReturn(CallExpr* call) {
   return retval;
 }
 
-static void insertRetMove(FnSymbol* fn, VarSymbol* retval, CallExpr* ret) {
+static bool hasGenericArrayReturn(FnSymbol* fn) {
+  if (returnsArray(fn)) {
+    BlockStmt* typeExpr = fn->retExprType;
+
+    CallExpr* call = toCallExpr(typeExpr->body.tail);
```

Comment:
What is this call supposed to be ? I'd INT_ASSERT it's the right thing or at least put a comment about what you are expecting